bridge/slack/helpers.go

```diff
@@ -58,58 +61,52 @@ func (b *Bslack) getChannelByID(ID string) (*slack.Channel, error) {
 	return nil, fmt.Errorf("%s: channel %s not found", b.Account, ID)
 }
 
-func (b *Bslack) getChannelID(name string) string {
-	idcheck := strings.Split(name, "ID:")
-	if len(idcheck) > 1 {
-		return idcheck[1]
-	}
-	for _, channel := range b.channels {
-		if channel.Name == name {
-			return channel.ID
-		}
-	}
-	return ""
-}
+var (
+	mentionRE  = regexp.MustCompile(`<@([a-zA-Z0-9]+)>`)
+	channelRE  = regexp.MustCompile(`<#[a-zA-Z0-9]+\|(.+?)>`)
+	variableRE = regexp.MustCompile(`<!((?:subteam\^)?[a-zA-Z0-9]+)(?:\|@?(.+?))?>`)
+	urlRE      = regexp.MustCompile(`<(.*?)(\|.*?)?>`)
+)
```

Comment:
The best-practice is to instantiate regexps at package-load time as package-wide variables. This way any issue in a regular expression will provoke an immediate crash and can't creep in silently because of lack of testing.